### ISSUE TITLE
[Task]: Research around data model session

### ISSUE THREAD

**Issue by sumiat** [HHS/simpler-grants-gov#991]
### Summary

- Research/discussion around data model session with stakeholders like Narmada, Lucas, and Julius 
- We want to answer questions to get to alignment on scope of the opportunity and consider trade-offs

### Acceptance criteria

- [ ] Open questions of the technical specification doc
- [ ] Technical specification doc from #989 updated

**Comment by sumiat** [comment]
- meeting scheduled for next week
- can we reduce the number of people in that session to just required engineering folx - who understands the current database

**Comment by sumiat** [comment]
will close after follow up session with david isaac

**Comment by chouinar** [comment]
The major point we need to sort out is regarding the usage of synopsis/forecast now, and how we might adjust that going forward. Ideally we'd come up with an approach that allows us to merge the two as they're very similar. The existing search experience already has that "merge" behavior. As it stands now, here's some logic we received from David Isaac:
<img width="863" alt="Screenshot 2024-02-21 at 4 29 32 PM" src="https://github.com/HHS/simpler-grants-gov/assets/46358556/a9568835-0e77-4dea-b2c5-4bd91b5ecc4a">

The 5th scenario is the main one that I think needs more discussion as I would have thought that a synopsis would always take precedent over a forecast record once the synopsis has been posted. I know there are some expected use cases regarding a synopsis closing and a forecast re-appearing for the next year, but it feels like there are better ways of handling that approach.

A very rough current working idea I have is to combine the forecast/synopsis records into a summary table. However, an opportunity could actually have multiple summaries, although only one "active" one (for the purposes of the detail page, or search when not considering historical data).

(Caveat to this section, I am not a designer or front-end engineer) Looking ahead to a newer grantor UX, we'd allow them to set which summary is "active" in one of two ways, either by creating a new one with a posting date of today or earlier (most commonly modifying a "live" summary) or by setting a posting date in the future at which point we'll automatically make it active then.

In the short-term, we'd need to only support an automated approach to this, and need to come up with a way to convert existing synopsis/forecast data into this type of model. While there are a lot of scenarios where it actually will be easy to know which one to mark active, what happens when we have both a forecast and synopsis record after their posting date, where neither is closed/archived? What do we choose? What happens when a synopsis is modified in the existing system, how do we handle merging that in?

I've expanded the above diagram a bit to try and visualize this.
![Screenshot 2024-02-21 at 4 44 50 PM](https://github.com/HHS/simpler-grants-gov/assets/46358556/500b76d2-da78-4573-b862-f604d5a7e24b)

I've changed the opportunity status a bit to instead be active/closed/archived. Active encompasses both forecasted/posted from the current system, and I changed forecasted to a separate field.
